core/src/Revolution/Processors/Security/Access/Policy/GetList.php

```diff
@@ -57,20 +68,101 @@ public function initialize()
+                            $c->where([
+                                '`modAccessContext`.`target`' => $context
+                            ]);
+                        }
+                        break;
+                    case 'MODx.grid.UserGroupResourceGroup':
+                        $joinClass = modAccessResourceGroup::class;
+                        $joinAlias = 'modAccessResourceGroup';
+                        $resourceGroup = $this->getProperty('resourceGroup', '');
+                        if (!empty($resourceGroup)) {
+                            $c->where([
+                                '`modAccessResourceGroup`.`target`' => (int)$resourceGroup
+                            ]);
+                        }
+                        break;
+                    case 'MODx.grid.UserGroupCategory':
+                        $joinClass = modAccessCategory::class;
+                        $joinAlias = 'modAccessCategory';
+                        $category = $this->getProperty('category', '');
+                        if (!empty($category)) {
+                            $c->where([
+                                '`modAccessCategory`.`target`' => (int)$category
+                            ]);
+                        }
+                        break;
+                    case 'MODx.grid.UserGroupSource':
+                        $joinClass = modAccessMediaSource::class;
+                        $joinAlias = 'modAccessMediaSource';
+                        $source = $this->getProperty('source', '');
+                        if (!empty($source)) {
+                            $c->where([
+                                '`modAccessMediaSource`.`target`' => (int)$source
+                            ]);
+                        }
+                        break;
+                    case 'MODx.grid.UserGroupNamespace':
+                        $joinClass = modAccessNamespace::class;
+                        $joinAlias = 'modAccessNamespace';
+                        // Note that namespace pk is a string ('name' is the dataIndex), not an int id
+                        $namespace = $this->getProperty('namespace', '');
+                        if (!empty($namespace)) {
+                            $c->where([
+                                '`modAccessNamespace`.`target`' => $namespace
+                            ]);
+                        }
+                        break;
+                }
+                $c->innerJoin(
+                    $joinClass,
+                    $joinAlias,
+                    [
+                        '`' . $joinAlias . '`.`policy` = `modAccessPolicy`.`id`',
+                        '`' . $joinAlias . '`.`principal` = ' . $userGroup,
+                        '`' . $joinAlias . '`.`principal_class` = ' . $this->modx->quote(modUserGroup::class)
+                    ]
+                );
+                $c->groupby('`modAccessPolicy`.`id`');
+            }
+        }
```

Comment:
Have you considered to use an early return of `$c` instead of using a huge if statement block?